source/sas_compress.cpp

```diff
@@ -101,18 +171,16 @@ SAS::Compressor::Compressor()
 }
 
 /// Compressor destructor.  Terminates the zlib compressor.
-SAS::Compressor::~Compressor()
+ZlibCompressor::~ZlibCompressor()
 {
   deflateEnd(&_stream);
 }
 
 /// Compresses the specified string using the optional profile.
-std::string SAS::Compressor::compress(const std::string& s, const Profile* profile)
+std::string ZlibCompressor::compress(const std::string& s, std::string dictionary)
```

Comment:
It is, but all we need is the dictionary, so it felt simpler just to pass in the piece of information we needed.

I think this is similar to your suggestion at https://github.com/Metaswitch/sas-client/pull/61#discussion_r47219168 - I'd like to be consistent, and either always extract the necessary piece of information from the Profile and pass that to the compressor, or always pass in the Profile and let the compressor do the extracting. I don't feel strongly - I'll do the former unless you feel strongly.